server/controllers/profileController.js

```diff
@@ -0,0 +1,110 @@
+const Profile = require("../models/ProfileModel");
+const User = require("../models/User");
+const mongoose = require("mongoose");
+
+const toObjectId = mongoose.Types.ObjectId;
+
+// create new profile using given arguments
+
+exports.createProfile = async (req, res, next) => {
+  try {
+    const { firstName, lasttName, phone, address, gender, description } =
+      req.body;
+
+    const userId = req.user.id;
+    const user = await User.findById(userId);
+    if (!user) {
+      res.status(401).json({ message: "Not Authorized" });
+    }
+
+    if (
+      firstName === undefined ||
+      lasttName === undefined ||
+      phone === undefined ||
+      address === undefined ||
+      gender === undefined ||
+      description === undefined
+    ) {
+      res.status(401).json({ message: "Complete the form please" });
+    }
+    const data = {
+      userId: toObjectId(userId),
+      firstName: firstName.trim().toString(),
```

Comment:
Note: In the future, you can add the `trim` property when creating models

Eg:  `firstName: { type: String, trim: true }`